### PR TITLE
Add test showing duplicates do not steal their originals' attachments

### PR DESCRIPTION
**Pull request by dannyroberts**
##### SUMMARY
This is a test I made for https://dimagi-dev.atlassian.net/browse/SAAS-10295. (I originally thought it showed the opposite of what it does, which you can see my exuberance about here: https://github.com/dimagi/commcare-hq/pull/26869).

There are no behavior changes, just the test, but it's a quick one, and it seems worth it to add now that I wrote it.

### CODE REVIEW

**Review by snopoke — COMMENTED** [review]
This looks like a slightly different version of one we already have:
https://github.com/dimagi/commcare-hq/blob/9d8bf79b2c42df84d44621557d70cf8b024d847c/corehq/ex-submodules/couchforms/tests/test_edits.py#L79

Probably worth removing one. This one seems cleaner.

**Comment by dannyroberts** [comment]
@snopoke I think these two test slightly (but importantly) different things. This one tests duplicates with attachments, that one tests form edits with attachments. For duplicates, the new form is sidelined (has its id changed) and the original retains its place; for edits, the original is sidelined (has its id changed) and is replaced by the new form.

**Review by snopoke — APPROVED** [review]

File: corehq/form_processor/tests/test_basics.py [hunk 1]
```diff
@@ -447,6 +449,34 @@ def test_globally_unique_case_id(self):
...
+            form_id = uuid.uuid4().hex
+            form = get_simple_form_xml(form_id)
+
+            form_1 = submit_and_fetch(form, DOMAIN, attachments)
+            self.assert_(not form_1.is_duplicate)
+            check_attachments(form_1, attachments)
+
+            form_1 = self.formdb.get_form(form_id)
+            form_2 = submit_and_fetch(form, DOMAIN, attachments)
+            self.assert_(not form_1.is_duplicate)
```
Comment (millerdev) [inline corehq/form_processor/tests/test_basics.py]: It looks like `form_1` is stale by the time this assertion is done. Would it make sense to fetch `form_1` after submitting `form_2` (i.e., swap line 464 and 465)?

Comment (dannyroberts) [reply corehq/form_processor/tests/test_basics.py]: Great point, thank you: https://github.com/dimagi/commcare-hq/pull/26895

**Pull request merged**